framework/web/Session.php

```diff
@@ -229,13 +226,29 @@ public function setHasSessionId($value)
     }
 
     /**
-     * Gets the session ID.
-     * This is a wrapper for [PHP session_id()](http://php.net/manual/en/function.session-id.php).
+     * Gets a session ID from the request or null if there is no session ID in the request
+     * @return string|null
+     */
+    private function getRequestSessionIdInternal()
+    {
+        $name = $this->getName();
+        $request = Yii::$app->getRequest();
+        if (ini_get('session.use_cookies') && !empty($_COOKIE[$name])) {
+            return $_COOKIE[$name];
+        }
+        if (!ini_get('session.use_only_cookies') && ini_get('session.use_trans_sid')) {
+            return $request->get($name);
+        }
+        return null;
+    }
+
+    /**
+     * Gets the session ID even the session hasn't yet started
      * @return string the current session ID
      */
     public function getId()
     {
-        return session_id();
+        return session_id() ?: ($this->getRequestSessionIdInternal() ?: '');
```

Comment:
No it couldn't.
The only way PHP knows the session ID if the session wasn't yet started is the request (cookie or get param). If the session was started (and even closed after that) the `session_id()` will return current ID instead of a request parameter.